packages/flutter_test/lib/src/controller.dart

```diff
@@ -32,11 +170,15 @@ const String _defaultPlatform = kIsWeb ? 'web' : 'android';
 /// Concrete subclasses must implement the [pump] method.
 abstract class WidgetController {
   /// Creates a widget controller that uses the given binding.
-  WidgetController(this.binding);
+  WidgetController(this.binding, [SemanticsController? semantics])
+    : semantics = semantics ?? SemanticsController(binding);
 
   /// A reference to the current instance of the binding.
   final WidgetsBinding binding;
 
+  /// Provides access to testing and control methods for semantics.
```

Comment:
This could be expanded.

It should somehow mention "accessibility" to give people an idea what use cases may require accessing this controller.

Reply:
I've been trying to avoid the "accessibility" term in this work, as the semantics is more than that according to [it's own documentation](https://api.flutter.dev/flutter/widgets/Semantics-class.html). That said, I could either point to those docs, or follow the example there and list out the same use cases to expand on this documentation.

Do you have a preference on those or another idea?

Reply:
Expanded this to mention the same use cases as described by the `Semantics` class.